instrumentation/hibernate/hibernate-4.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/hibernate/v4_0/SessionTest.java

```diff
@@ -0,0 +1,880 @@
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(
+                                stringKey("hibernate.session_id"),
+                                trace
+                                    .getSpan(1)
+                                    .getAttributes()
+                                    .get(stringKey("hibernate.session_id")))),
+                span ->
+                    span.hasKind(CLIENT)
+                        .hasParent(trace.getSpan(3))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(SemanticAttributes.DB_SYSTEM, "h2"),
+                            equalTo(SemanticAttributes.DB_NAME, "db1"),
+                            equalTo(SemanticAttributes.DB_USER, "sa"),
+                            equalTo(SemanticAttributes.DB_CONNECTION_STRING, "h2:mem:"),
+                            satisfies(
+                                SemanticAttributes.DB_STATEMENT,
+                                val -> val.isInstanceOf(String.class)),
+                            satisfies(
+                                SemanticAttributes.DB_OPERATION,
+                                val -> val.isInstanceOf(String.class)),
+                            equalTo(SemanticAttributes.DB_SQL_TABLE, "Value"))));
+  }
+
+  private static Stream<Arguments> provideArgumentsHibernateReplicate() {
+    return Stream.of(
+        Arguments.of(
+            named(
+                "replicate",
+                new Parameter(
+                    "replicate",
+                    (Session session, Value val) -> {
+                      Value replicated = new Value(val.getName() + " replicated");
+                      replicated.setId(val.getId());
+                      session.replicate(replicated, ReplicationMode.OVERWRITE);
+                    },
+                    null))),
+        Arguments.of(
+            named(
+                "replicate",
+                new Parameter(
+                    "replicate",
+                    (Session session, Value val) -> {
+                      Value replicated = new Value(val.getName() + " replicated");
+                      replicated.setId(val.getId());
+                      session.replicate(
+                          "io.opentelemetry.javaagent.instrumentation.hibernate.v4_0.Value",
+                          replicated,
+                          ReplicationMode.OVERWRITE);
+                    },
+                    null))));
+  }
+
+  @Test
+  void testHibernateFailedReplicate() {
+    testing.runWithSpan(
+        "parent",
+        () -> {
+          Session session = sessionFactory.openSession();
+          session.beginTransaction();
+
+          try {
+            session.replicate(123L /* Not a valid entity */, ReplicationMode.OVERWRITE);
+          } catch (RuntimeException e) {
+            // We expected this, we should see the error field set on the span.
```

Comment:
Nit: could a `MappingException` be caught here (e.g. via assertj's `catchThrowable()`) and passed to the assertions?